tmt/steps/provision/__init__.py

```diff
@@ -449,32 +449,39 @@ def _ansible_playbook_path(self, playbook):
         self.debug(f"Playbook full path: '{playbook}'", level=2)
         return playbook
 
-    def _export_environment(self, execute_environment=None):
-        """ Prepare shell export of environment variables """
+    def _prepare_environment(self, execute_environment=None):
+        """ Prepare dict of environment variables """
         # Prepare environment variables so they can be correctly passed
         # to ssh's shell. Create a copy to prevent modifying source.
         environment = dict()
         environment.update(execute_environment or dict())
         # Plan environment and variables provided on the command line
         # override environment provided to execute().
         environment.update(self.parent.plan.environment)
-        # Prepend with export and run as a separate command.
-        if not environment:
-            return ''
-        return 'export {}; '.format(
-            ' '.join(tmt.utils.shell_variables(environment)))
+        return {} if not environment else environment
+
+    @staticmethod
+    def _export_environment(environment):
+        """ Prepare shell export of environment variables """
+        return f'export {" ".join(tmt.utils.shell_variables(environment))}; ' \
+            if environment \
+            else ''
 
     def ansible(self, playbook, extra_args=None):
         """ Prepare guest using ansible playbook """
         playbook = self._ansible_playbook_path(playbook)
+        self.run(f'stty cols {tmt.utils.OUTPUT_WIDTH};:',
+                 shell=True)
+        ssh_common_args = [f'--ssh-common-args="{i}"'
+                           for i in self._ssh_options(join=True).split()]
```

Comment:
One more thought here: Would it work to include all the common ssh options in a single ansible option?

    f"--ssh-common-args='{self._ssh_options(join=True)}'"

This would also remove the need to [remove space](https://github.com/teemtee/tmt/pull/977/files#diff-f684593942660856c3bdc810b597c3956b4107716a554eece2c2bf08344a9503L293) from all ssh options.

Reply:
Thanks @lukaszachy , your proposal works.

> Would it work to include all the common ssh options in a single ansible option?
No it doesn't work. I've already tried with the single ansible option:
```
                out: TASK [Gathering Facts] *********************************************************
                out: task path: /var/tmp/tmt/run-024/plans/features/advanced/discover/default/tests/tests/prepare/ansible/data/playbook.yml:1
                out: <root@127.0.0.1> ESTABLISH SSH CONNECTION FOR USER: None
                out: <root@127.0.0.1> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 '-oForwardX11=no -oStrictHostKeyChecking=no -oUserKnownHostsFile=/dev/null -oIdentitiesOnly=yes -p10025 -i/var/tmp/tmt/run-025/plan/provision/default/id_ecdsa -S/run/user/1000/tmt/tmps4tlevco' -o ControlPath=/home/dkarpele/.ansible/cp/1d20ae1049 root@127.0.0.1 '/bin/sh -c '"'"'echo ~ && sleep 0'"'"''
                out: <root@127.0.0.1> (255, b'', b'command-line line 0: garbage at end of line; "-oStrictHostKeyChecking".\r\n')
                out: fatal: [root@127.0.0.1]: UNREACHABLE! => {
                out:     "changed": false,
                out:     "msg": "Failed to connect to the host via ssh: command-line line 0: garbage at end of line; \"-oStrictHostKeyChecking\".",
                out:     "unreachable": true
                out: }
                out: 
                out: PLAY RECAP *********************************************************************
                out: root@127.0.0.1             : ok=0    changed=0    unreachable=1    failed=0    skipped=0    rescued=0    ignored=0
                out: 
                out: Failed to run command 'ansible-playbook -vvv --ssh-common-args="-oForwardX11=no -oStrictHostKeyChecking=no -oUserKnownHostsFile=/dev/null -oIdentitiesOnly=yes -p10025 -i/var/tmp/tmt/run-025/plan/provision/default/id_ecdsa -S/run/user/1000/tmt/tmps4tlevco" -i root@127.0.0.1, /var/tmp/tmt/run-024/plans/features/advanced/discover/default/tests/tests/prepare/ansible/data/playbook.yml'. Reason: Command returned '4'.
```
```

Reply:
> No it doesn't work. I've already tried with the single ansible option:

Ok, then we don't have a choice :)